ginga/misc/Future.py

```diff
@@ -24,7 +27,8 @@ def __init__(self, data=None):
 
     def get_data(self):
         return self.data
-    
+
+    # TODO: Could add some args type/value, return value validation here
```

Comment:
@ejeschke Yes, I understand. I meant like we can check when adding the method in freeze() that if it is really a `callable`. Also, in other places we are checking on the value returned by the method. So, we may want to see if it returns a non-null value. I may be over-killing this, but in my opinion Python being a dynamically-typed language, we would explicitly need to protect ourselves against things like this, especially in an open-source project. This class was created for a specific purpose and serves the author's purposes well for now. But we never know and can barely imagine all scenarios about how a new contributor would end up using it. So my point with this, we should add protection agains all undesirable scenarios if and when we can.